backend-schedule/src/main/java/eu/socialedge/hermes/backend/schedule/domain/api/ScheduleGenerationApi.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Hermes - The Municipal Transport Timetable System
+ * Copyright (c) 2017 SocialEdge
+ * <p>
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ */
+package eu.socialedge.hermes.backend.schedule.domain.api;
+
+import eu.socialedge.hermes.backend.schedule.domain.BasicScheduleGenerator;
+import eu.socialedge.hermes.backend.schedule.domain.Schedule;
+import eu.socialedge.hermes.backend.schedule.domain.ScheduleSpecification;
+import eu.socialedge.hermes.backend.schedule.domain.exception.NotFoundException;
+import eu.socialedge.hermes.backend.transit.domain.Route;
+import eu.socialedge.hermes.backend.transit.domain.repository.RouteRepository;
+import lombok.val;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import javax.validation.Valid;
+import javax.validation.constraints.NotNull;
+
+import java.util.Optional;
+
+import static org.springframework.web.bind.annotation.RequestMethod.POST;
+
+@RestController
+@RequestMapping(path = "/schedules")
```

Comment:
Agree with highlighting, but I think that /generators will confuse, because we are not working with some generator resources, so I prefer to have it as noun, e.g. 'generate'. It points that we are working with schedules and doing a generation of it